PauloAlves8039/dotnet-parking-backend
# [FEATURE] Implement DTO Classes and Mapping Profiles

### Context
This Pull Request introduces Data Transfer Objects (DTOs) and a mapping profile to facilitate the conversion between domain models and DTOs. This structured approach improves data handling and validation in the application.

### Changes
DTO Classes: Created the following DTO classes for data transfer:

- CustomerDTO
- AddressDTO
- CustomerVehicleDTO
- VehicleDTO
- StayDTO

Mapping Profile: Added `ParkingProfile` class to handle mappings between domain models and DTOs.

- Mappings between `Customer` and `CustomerDTO`
- Mappings between `Address` and `AddressDTO`
- Mappings between `CustomerVehicle` and `CustomerVehicleDTO`
- Mappings between `Vehicle` and `VehicleDTO`
- Mappings between `Stay` and `StayDT`O
- Validations: Added validation logic to ensure data integrity.
- Migrations: Updated fields and database schema to reflect the changes.

### How to Test
1. Configure local DB settings: Ensure your local database settings are properly configured.
2. Run migrations: Execute `dotnet ef database update` to apply the latest database migrations.
3. Verify DTOs: Test the DTOs to ensure they are correctly receiving and sending data.
4. Validate mappings: Ensure the mappings between domain models and DTOs are accurate and functioning as expected.
5. Check validations: Test the validation logic to ensure it properly handles data constraints.

### Notes
- Backup your current database before applying new migrations to prevent data loss.
- Review the naming conventions for consistency throughout the codebase.
- Thoroughly test all mappings and DTOs to ensure data integrity and correct functionality.

### Related Issues
- Link to related issues or tasks, if applicable.
